XLS-0033d-CFTs/README.md

```diff
@@ -514,35 +515,72 @@ We propose using the following format for CFT amounts::
 
 ```json
 "amount": {
-  "issuer": "rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn",
-  "cft_asset": "USD",
+  "cft_issuance_id": "00070C4495F14B0E44F78A264E41713C64B5F89242540EE255534400000000000000",
   "value": "1"
 }
 ```
 
-The idea behind this format is that it adds only one new subfield to the `amount` field, but still distinguishes between itself and Issued Currency amounts. Using the CFT ID directly would not allow us to easily find the underlying `CFTokenIssuance` object because we would still need the `issuer` to know where to look for it.
+Since the `AssetCode` is an optional field, the `CFTokenIssuanceID` will be used to uniquely identify the CFT during a Payment transaction.
+
+### 1.3.5 The **`CFTokenTrust`** Transaction
+This transaction allows a holder to own a `CFToken`, creating a `CFToken` object for the holder with zero balance. 
+
+If the issuer has set `lsfRequiresAuthorization`(allowlisting) on the `CFTokenIssuance`, then the issuer must submit a `CFTokenTrust` transaction as well in order to give permission to the holder. If `lsfRequiresAuthorization` is not set and the issuer attempts to submit this transaction, it will fail. Read more about allowlisting in Section 1.8.0. 
+#### 1.3.5.1 CFTokenTrust
+| Field Name      | Required?          | JSON Type | Internal Type |
+| --------------- | ------------------ | --------- | ------------- |
+| `Account`       | :heavy_check_mark: | `string`  | `ACCOUNTID`   | 
+
+This address can indicate either an issuer or a potential holder of a CFT.
+
+| Field Name         | Required? | JSON Type | Internal Type |
+| ------------------ | --------- | --------- |---------------|
+| `TransactionType`  | ️ ✔        | `object`  | `UINT16`      |
+
+Indicates the new transaction type **`CFTokenTrust`**. The integer value is `29 (TODO)`.
+
+| Field Name  | Required? | JSON Type | Internal Type |
+| ----------- | --------- | --------- | ------------- |
+| `CFTokenIssuanceID` |  ✔️  | `string`  | `UINT256`     | 
+
+Indicates the ID of the CFT involved. 
+
+| Field Name  | Required? | JSON Type | Internal Type |
+| ----------- | --------- | --------- | ------------- |
+| `CFTokenHolder` |    | `string`  | `ACCOUNTID`     | 
 
-### 1.3.5 The **`AccountDelete`** Transaction
+This field is only used for authorization/allowlisting. It's an optional field that specifies the holder's address whom the issuer wants to authorize.
```

Comment:
> Do we need both Account and CFTokenHolder fields in this CFTokenTrust transaction?

`Account` field is a mandatory field in transaction I believe. `CFTokenHolder` will always be holder's address, it will only be used when issuer wants to authorize a holder.

> As an alternative design: If we only used `CFTokenHolder`, then this could always specify what it says -- the holder address. Here we can consider the following scenarios:
> 
> 1. Issuer is authorizing a holder (single-sig): The transactor looks-up the `CFTokenIssuanceID` and finds the issuer address from there (i.e., the `Issuer` field). Signed by the issuer.
> 2. Issuer is authorizing a holder (multi-sig): Same as above (signed by issuer quorum)
> 3. Holder is authorizing an issuer (single-sig): Same as above (must be signed by holder)
> 4. Holder is authorizing an issuer (multi-sig): Same as above (signed by holder quorum).

Allowlisting has nothing to do with with multisigning (unless you are suggesting otherwise)

To clear things up, there are two main scenario how CFTokenTrust will be used:

### Example
Assume that Alice is the issuer of USD, and Bob wants to use it.

#### If USD doesn't have allowlisting:
1. Bob sends a CFTokenTrust to hold Alice's token like:
```
{
  TransactionType: "CFTokenTrust",
  Account: Bob,
  CFTokenIssuanceID: "123",
}
```
This creates a zero-balance `CFToken` object for Bob, and now Bob can receive USD from anybody. Note here, that `CFTokenHolder` is not specified, because Bob's address(holder) is in the `Account` field

#### If USD has allowlisting enabled:
1. (Same as above)Bob sends a CFTokenTrust to hold Alice's token like:
```
{
  TransactionType: "CFTokenTrust",
  Account: Bob,
  CFTokenIssuanceID: "123",
}
```

2. Alice(issuer) sends a CFTokenTrust to authorize Bob:
```
{
  TransactionType: "CFTokenTrust",
  Account: Alice,
  CFTokenIssuanceID: "123",
  CFTokenHolder: Bob
}
```
If this transaction is successful, it will set the `tfAuthorized` flag on Bob's CFToken.
Here, we need both `Account` and `CFTokenHolder` because Alice needs to specify which holder's account she wants to authorize, in case, she want's to authorize Bob to use USD.

Reply:
Resolving this to unblock for now. I'll open another PR after I think through this some more (to decided if it's still an issue).